app/models/user.rb

```diff
@@ -408,15 +492,35 @@ def find_unique_new_barcode
     # a) pick barcode in MLN db
     # b) make sure it'd be new and unique in Sierra
     # c) repeat until b) is true
+    LogWrapper.log('DEBUG', {
+       'message' => "Checking user #{self.id || 'NIL'}",
+       'method' => "#{model_name}.find_unique_new_barcode",
+       'status' => "start"
+      })
 
-    # TODO: future branches: move user.assign_barcode over to its own multithreaded job,
-    # and move save_as_complete over to after have successfully sent to Sierra
-    self.assign_barcode!
-    self.save_as_complete!
+    # Enqueue a job to be performed as soon as the queuing system is free.
+    begin
+      # Note: user.pin is not getting serialized properly, hence passing as its own var
```

Comment:
Please add into this comment that `pin` is not persisted in the database, so it is not possible to retrieve it later

Reply:
done